lib/double_entry/transfer.rb

```diff
@@ -72,34 +71,52 @@ def initialize(attributes)
       end
     end
 
-    def process(amount, from, to, code, detail)
-      if from.scope_identity == to.scope_identity && from.identifier == to.identifier
-        fail TransferNotAllowed, 'from and to are identical'
+    def process(amount, options)
+      from_acct = options[:from]
+      to_acct = options[:to]
+      code = options[:code]
+      detail = options[:detail]
+      metadata = options[:metadata]
+      if from_acct.scope_identity == to_acct.scope_identity && from_acct.identifier == to_acct.identifier
+        fail TransferNotAllowed, 'from_acct and to_acct are identical'
       end
-      if to.currency != from.currency
-        fail MismatchedCurrencies, "Missmatched currency (#{to.currency} <> #{from.currency})"
+      if to_acct.currency != from_acct.currency
+        fail MismatchedCurrencies, "Mismatched currency (#{to_acct.currency} <> #{from_acct.currency})"
       end
-      Locking.lock_accounts(from, to) do
-        credit, debit = Line.new, Line.new
+      Locking.lock_accounts(from_acct, to_acct) do
+        credit, debit = create_lines(amount, code, detail, from_acct, to_acct)
+        create_line_metadata(credit, debit, metadata) if metadata
+      end
+    end
+
+    def create_lines(amount, code, detail, from_acct, to_acct)
+      credit, debit = Line.new, Line.new
 
-        credit_balance = Locking.balance_for_locked_account(from)
-        debit_balance  = Locking.balance_for_locked_account(to)
+      credit_balance = Locking.balance_for_locked_account(from_acct)
+      debit_balance  = Locking.balance_for_locked_account(to_acct)
 
-        credit_balance.update_attribute :balance, credit_balance.balance - amount
-        debit_balance.update_attribute :balance, debit_balance.balance + amount
+      credit_balance.update_attribute :balance, credit_balance.balance - amount
+      debit_balance.update_attribute :balance, debit_balance.balance + amount
 
-        credit.amount,  debit.amount  = -amount, amount
-        credit.account, debit.account = from, to
-        credit.code,    debit.code    = code, code
-        credit.detail,  debit.detail  = detail, detail
-        credit.balance, debit.balance = credit_balance.balance, debit_balance.balance
+      credit.amount, debit.amount   = -amount, amount
+      credit.account, debit.account = from_acct, to_acct
+      credit.code, debit.code       = code, code
+      credit.detail, debit.detail   = detail, detail
+      credit.balance, debit.balance = credit_balance.balance, debit_balance.balance
```

Comment:
i know this was just an indentation change, so feel free to completely ignore this comment: Im not a fan of this style of initialisation